src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -152,28 +184,42 @@ def _add_layer(self, con_type_data: ConnectionTypeData, connections: List[List[i
             for conn in conns:
                 known_strength = known_strengths.get((conn, agent.index), None)
                 if known_strength != None:
-                    strengthes[np.where(conns==conn)] = known_strength
+                    strengthes[np.where(conns == conn)] = known_strength
                     del known_strengths[(conn, agent.index)]
                 else:
                     # connection is new. store the strength for future use
-                    known_strengths[(agent.index, conn)] = strengthes[np.where(conns==conn)]
+                    known_strengths[(agent.index, conn)] = strengthes[np.where(conns == conn)]
+
+            daily_connections = [conn for conn in conns if strengthes[np.where(conns == conn)] ==
+                                 con_type_data.connection_strength]
+            weekly_connections = [conn for conn in conns if strengthes[np.where(conns == conn)] !=
```

Comment:
If it's daily_connections 's complement, then:
* if conn not in daily_connections  would be more appropriate, now?
* Why you  need to hold to weekly connections?